mirdata/datasets/saraga_carnatic.py

```diff
@@ -347,39 +354,39 @@ def load_tempo(tempo_path):
 
     tempo_annotation = {}
 
-    with open(tempo_path, 'r') as reader:
-        parsed_tempo = reader.readline()
-
-        tempo_data = []
-        tempo_apm = parsed_tempo.split(',')[0]
-        tempo_data.append(tempo_apm)
-        tempo_bpm = parsed_tempo.split(',')[1].split(' ')[1]
-        tempo_data.append(tempo_bpm)
-        sama_interval = parsed_tempo.split(',')[2].split(' ')[1]
-        tempo_data.append(sama_interval)
-        beats_per_cycle = parsed_tempo.split(',')[3].split(' ')[1]
-        tempo_data.append(beats_per_cycle)
-        subdivisions = parsed_tempo.split(',')[4].split(' ')[1]
-        tempo_data.append(subdivisions)
-
-        if 'NaN' in tempo_data:
-            return None
-
-        tempo_annotation['tempo_apm'] = (
-            float(tempo_apm) if '.' in tempo_apm else int(tempo_apm)
-        )
-        tempo_annotation['tempo_bpm'] = (
-            float(tempo_bpm) if '.' in tempo_bpm else int(tempo_bpm)
-        )
-        tempo_annotation['sama_interval'] = (
-            float(sama_interval) if '.' in sama_interval else int(sama_interval)
-        )
-        tempo_annotation['beats_per_cycle'] = (
-            float(beats_per_cycle) if '.' in beats_per_cycle else int(beats_per_cycle)
-        )
-        tempo_annotation['subdivisions'] = (
-            float(subdivisions) if '.' in subdivisions else int(subdivisions)
-        )
+    with open(tempo_path, 'r') as fhandle:
+        reader = csv.reader(fhandle, delimiter=',')
+        for line in reader:
+            tempo_data = []
+            tempo_apm = line[0]
+            tempo_data.append(tempo_apm)
+            tempo_bpm = line[1]
+            tempo_data.append(tempo_bpm)
+            sama_interval = line[2]
+            tempo_data.append(sama_interval)
+            beats_per_cycle = line[3]
+            tempo_data.append(beats_per_cycle)
+            subdivisions = line[4]
+            tempo_data.append(subdivisions)
+
+            if 'NaN' in tempo_data:
+                return None
```

Comment:
Bonus, if you feel like adding one more test to `test_saraga_carnatic.py` to check this case!

Reply:
Cool! I forgot that one. I'll do it together with the rest of the fixes. Thanks for the catch!